### DISCUSSION TITLE
Mismatch in timing between fit.summary() and leaderboard() functions

### DISCUSSION THREAD

**Discussion by praveentitusf** [autogluon/autogluon#4759] [Q&A] [answered]
Question 1: Why there is a mismatch in timing for "WeightedEnsemble_L2 model" between fit.summary()['model_fit_times']  and fit_time in leaderboard functions, as shown in the images.  what does the fit_time  = 1475.84 in leaderboard table mean? 

From reading this https://auto.gluon.ai/stable/api/autogluon.tabular.TabularPredictor.leaderboard.html
what I understood in leaderboard  
fit_time = fit_time_marginal + time taken to fit base models = 1583.696767

could someone explain the mismatch.

Q2) Is the timing unit is seconds?

Q3) Is there a easy way to find the total runtime after fitting all models. 

**code**: predictor.fit_summary()['model_fit_times']
**output**:
![image](https://github.com/user-attachments/assets/bbfd4e23-75a0-4731-9c0e-53debb9a09c3)

**Code**: predictor.leaderboard()
output:
![image](https://github.com/user-attachments/assets/6affade7-3fbf-4372-afe8-c9a4b5a1c329)

**Comment by Innixma** [comment, marked as answer]
In leaderboard it is the sum of all component models that are required for the Weighted Ensemble model to work at test time.

For example, CatBoost is used by the weighted ensemble, so its `1407.02` second `fit_time_marginal` is added to the weighted ensemble's `fit_time`.

To see the matching value as in `predictor.fit_summary()`, refer to `fit_time_marginal`.
